### ISSUE TITLE
question about softmax

### ISSUE THREAD

**Issue by Continue7777** [jadore801120/attention-is-all-you-need-pytorch#52]
when calulate the attention in MultiHead ,softmax's dim is 0,but i think dim=2 is right.

**Comment by jshi31** [comment]
the softmax has default dim -1, so in the code the dim=2

**Comment by jadore801120** [comment]
Hello @Continue7777 , thanks for the observation. You are right, set the dimension explicitly is the best practice.
@jshi31 Thanks for the explanation.
The code has been updated to 0.4.1, please have a look.
Thanks,
Yu-Hsiang
